File: src/main/java/chat/teco/tecochat/chat/domain/keyword/Keyword.java
package chat.teco.tecochat.chat.domain.keyword;

import chat.teco.tecochat.domain.chat.Chat;
import chat.teco.tecochat.support.domain.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class Keyword extends BaseEntity {

    private String keyword;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_id")
    private Chat chat;

    protected Keyword() {
    }

    public Keyword(String keyword, Chat chat) {
        this.keyword = keyword;
        this.chat = chat;
    }

    public String keyword() {
        return keyword;
    }

    public Chat chat() {
        return chat;
    }

    @Override
    public String toString() {
        return "Keyword=[" + keyword + "]";
    }

    public Keyword copy(Chat copiedChat) {
        return new Keyword(keyword, copiedChat);
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/keyword/Keyword.kt
package chat.teco.tecochat.domain.keyword

import chat.teco.tecochat.domain.chat.Chat
import chat.teco.tecochat.support.domain.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class Keyword(
    var keyword: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_id")
    var chat: Chat,

    id: Long = 0L,
) : BaseEntity(id) {

    constructor(keyword: String, chat: Chat) : this(
        keyword = keyword,
        chat = chat,
        id = 0L
    )

    fun copy(copiedChat: Chat): Keyword {
        return Keyword(keyword, copiedChat)
    }
}
